### PR TITLE
Allow for overriding RAW endpoint usage page and ID.

### PR DESCRIPTION
**Pull request by tzarc**
## Description

Adds support for overriding the RAW endpoint's usage page and ID.
Sanity checked by compiling `make all:via` -- no errors.
We have #7640 in the pipeline for documenting the RAW endpoint, will wait for that to merge before suggesting appropriate changes to the documentation.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
What's the use case for this?

File: quantum/usb_descriptor_common.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/* Copyright 2020 Nick Brassel (tzarc)
```
Comment (zvecr) [inline quantum/usb_descriptor_common.h:1]: Any chance we move this to `tmk_core/protocol/usb_descriptor_common.h`?

Comment (tzarc) [reply quantum/usb_descriptor_common.h:1]: Done.

**Comment by tzarc** [comment]
> 
> 
> What's the use case for this?

Purely configurability -- I had thought about this a while back when considering dual-bank flash and being able to flash the kb _while the keyboard is active_ -- at the time it seemed like it might be safer to have a separate usage page. Not planning to do the flashing now, but figured the configurability might be useful.

**Comment by tzarc** [comment]
Will wait for the RAW HID doco PR #7640 to go through, first.

**Pull request merged**